### PR TITLE
Improve scheduler & update dependenies

### PR DESCRIPTION
**Pull request by zjb0807**

### CODE REVIEW

**Review by xlc — APPROVED** [review]

File: scheduler/test/Scheduler.test.ts [hunk 1]
```diff
@@ -76,21 +76,15 @@ describe("Schedule", () => {
...
-      await schedule.cancelCall(ethers.utils.hexlify(decode_log.args.task_id));
-    } else {
-      expect(false).to.not.be.ok;
-    }
+
+    let event = data.events.filter(item => provider.api.events.evm.Log.is(item.event));
+    expect(event.length).to.equal(1);
+
+    let decode_log = await iface.parseLog(event[0].event.data.toJSON()[0]);
+    console.log("task_id:" + decode_log.args.task_id);
```
Comment (ThunderDeliverer) [inline scheduler/test/Scheduler.test.ts:86]: Could we remove the console.log from tests? I think that the example description should be enough when running the tests.

File: scheduler/test/Scheduler.test.ts [hunk 2]
```diff
@@ -76,21 +76,15 @@ describe("Schedule", () => {
...
+    await schedule.cancelCall(ethers.utils.hexlify(decode_log.args.task_id));
```
Comment (ThunderDeliverer) [inline scheduler/test/Scheduler.test.ts:87]: We could do a validation that the `CancelledCall` is emitted, which would also mean, that the `cancelCall` is executed as expected.

```suggestion
    await expect(schedule.cancelCall(ethers.utils.hexlify(decode_log.args.task_id)))
       .to.emit(schedule, "CancelledCall")
       .withArgs(await wallet.getAddress(), ethers.utils.hexlify(decode_log.args.task_id)));
```

File: scheduler/test/Scheduler.test.ts [hunk 3]
```diff
@@ -103,20 +97,15 @@ describe("Schedule", () => {
...
-    } else {
-      expect(false).to.not.be.ok;
-    }
+
+    let event = data.events.filter(item => provider.api.events.evm.Log.is(item.event));
+    expect(event.length).to.equal(1);
+
+    let decode_log = await iface.parseLog(event[0].event.data.toJSON()[0]);
+    console.log("task_id:" + decode_log.args.task_id);
+    await schedule.rescheduleCall(5, ethers.utils.hexlify(decode_log.args.task_id));
```
Comment (ThunderDeliverer) [inline scheduler/test/Scheduler.test.ts:108]: I'd add an additional check here as well:

```suggestion
    await expect(schedule.rescheduleCall(5, ethers.utils.hexlify(decode_log.args.task_id)))
      .to.emit(schedule, "RescheduledCall")
      .withArgs(await wallet.getAddress(), ethers.utils.hexlify(decode_log.args.task_id));
```

**Review by ThunderDeliverer — COMMENTED** [review]
Everything looks great, but I have a couple of suggestions. 👌

**Pull request merged**